index.js

```diff
@@ -289,61 +346,83 @@ const legacyScriptingRunner = (Zap, zobj, app) => {
       }
 
       // Running "full" scripting method like KEY_poll
-      promise = runEvent({ key, name: fullEventName }, zobj, bundle);
+      result = await runEvent({ key, name: fullEventName }, zobj, bundle);
     } else {
       const preMethod = preMethodName ? Zap[preMethodName] : null;
-      if (preMethod) {
-        promise = runEvent({ key, name: preEventName }, zobj, bundle);
-      } else {
-        promise = Promise.resolve(bundle.request);
+      const request = preMethod
+        ? await runEvent({ key, name: preEventName }, zobj, bundle)
+        : bundle.request;
+
+      const isBodyStream = typeof _.get(request, 'body.pipe') === 'function';
+
+      if (hasFileFields(bundle) && !isBodyStream) {
+        const data = {};
```

Comment:
Getting a little confused on the call flow here. If we executed `runEvent()` above, then we will have used `parseFinalResult`, which will have already performed logic similar to what comes below with setting `request.body`. So is this block for when there was no `_pre` method defined?

Reply:
Yes, it is for when there's no `KEY_pre_` method. Pushed 3fbeddc to make it more clear.

Reply:
Cool, that helps. Reading over each of those functions, Is it possible to combine them into one? I see the `addFilesToRequestBodyFromPreResult` has extra work to do because the dev may have swapped up or added files to the `request` object. If we ran a request through that same logic that hadn't been modified with a `pre` method, would it have the same result?